src/vncproxy.ml

```diff
@@ -62,42 +62,58 @@ let exn_to_string = function
   | e -> Printexc.to_string e
 
 let find_console rpc session_id x =
+  Lwt_io.eprintlf "Trying to find console for VM %s" x >>= fun () ->
   (* Treat 'x' first as a uuid and then as a name_label *)
   Lwt.catch
     (fun () -> VM.get_by_uuid ~rpc ~session_id ~uuid:x )
     (fun  _ ->
        VM.get_by_name_label ~rpc ~session_id ~label:x >>= function
-       | [ exactly_one ] -> return exactly_one
+       | [ exactly_one ] -> Lwt.return exactly_one
        | [] ->
-         Printf.fprintf stderr "Failed to find VM %s\n%!" x;
-         fail (Failure (Printf.sprintf "Failed to find VM %s" x))
+         Lwt_io.eprintlf "Failed to find VM %s" x >>= fun () ->
+         Lwt.fail_with (Printf.sprintf "Failed to find VM %s" x)
        | y :: _ ->
          VM.get_uuid ~rpc ~session_id ~self:y >>= fun uuid ->
-         Printf.fprintf stderr "More than one VM found with this name.\n%!";
-         Printf.fprintf stderr "Choosing VM with uuid %s.\n%!" uuid;
-         return y
-
-    ) >>= fun vm ->
+         Lwt_io.eprintlf "More than one VM found with this name" >>= fun () ->
+         Lwt_io.eprintlf "Choosing VM with uuid %s" uuid >>= fun () ->
+         Lwt.return y
+    )
+  >>= fun vm ->
   (* Check that the VM is running *)
-  VM.get_power_state ~rpc ~session_id ~self:vm >>= fun power_state ->
+  VM.get_power_state ~rpc ~session_id ~self:vm
+  >>= fun power_state ->
   (if power_state <> `Running then begin
-      Printf.fprintf stderr "The VM %s is not running.\n%!" vm;
-      fail (Failure (Printf.sprintf "The VM is not running"))
-    end else return_unit) >>= fun () ->
+      Lwt_io.eprintlf "The VM %s is not running" vm >>= fun () ->
+      Lwt.fail_with "The VM is not running"
+    end else Lwt.return_unit) >>= fun () ->
 
   (* Find a console with the RFB protocol *)
-  VM.get_consoles ~rpc ~session_id ~self:vm >>=
-  Lwt_list.map_s (fun c -> Console.get_record ~rpc ~session_id ~self:c) >>= fun all ->
-  match List.filter (fun c -> c.API.console_protocol = `rfb) all with
-  | [ exactly_one ] -> return exactly_one
+  VM.get_consoles ~rpc ~session_id ~self:vm >>= fun cs ->
+  Lwt_list.filter_map_s (fun c ->
+      Lwt.catch
+        (fun () ->
+           Console.get_record ~rpc ~session_id ~self:c
+           >>= fun c_record ->
+           if c_record.API.console_protocol = `rfb
+           then Lwt.return_some c_record
+           else Lwt.return_none)
+        (fun e ->
+          (* Lwt_io.eprintlf "warning: %s" (Printexc.to_string e) >>= fun () -> *)
+          Lwt.return_none)
+    ) cs >>= fun all -> 
+  match all with
+  | [ exactly_one ] -> Lwt.return exactly_one
   | [] ->
-    Printf.fprintf stderr "The VM is exposing no VNC consoles.\n%!";
-    fail (Failure (Printf.sprintf "The VM is exposing no VNC consoles"))
```

Comment:
It might be useful to extract this `eprintlf` + `fail_with` pattern into a helper function in a future PR.

Reply:
We could probably simplify the code further in many places, but I did not want to spend too much time (it took already long enough to pinpoint what was failing)